CleverRaven/Cataclysm-DDA
# Flexible Per Pocket Encumbrance

#### Summary
Features "Pockets and armor can now specify a modifier on the 1 encumbrance per 250ml of volume"


#### Purpose of change
With the addition of more complex ablative vests, ablative pockets were contributing encumbrance twice to characters. Once from their fixed additional encumbrance but on items without a fixed max encumbrance their pockets were contributing there as well.

So I was going to just make a hard coded "if ablative skip" check but decided to make it more open instead so any pocket and armor data entry can have a modifier on the max encumbrance based on their volume.



#### Describe the solution

- Unharcoded the 250ml per encumbrance value. It now lives as a constant in the armor value
- Pockets and armor_data entries can have a modifier "volume_encumber_modifier" which scales the max encumbrance of the item based on volume. In the case of pockets it also modifies how much that specific pocket contributes to the relative "fullness" of the item. Ex: if you had 2 1L pockets and one had .5 volume_encumber_modifier your max encumbrance would be 6 (1L + (1L *.5) / 250mL) but if the better balanced pocket was full it would contribute only 2 to the max encumber not 3. 
- Ablative pockets have a modifier of 0 (meaning they don't contribute to max encumbrance). They still contribute in their normal way
- Added this setup to the scavenger armor to test it. It required **VERY** small scaling values to get it anywhere near its old values



#### Describe alternatives you've considered



#### Testing

- Load up the game check out Ballistic vests and see that they, even though they no longer have a clamped maxed encumbrance, still have a fixed encumbrance.
- Check that the Scavenger Armor has reasonable encumbrance values
- Put a Hub mantle in a Hub 01 Modular Defense System. Notice the encumbrance only goes up by the 5 it should

Here's a scavenger armor FILLED with meat
![image](https://user-images.githubusercontent.com/4514073/147880849-7e49a0f3-5ba0-4dbf-950a-bfd96c5f3cdc.png)




#### Additional context
Scavenger armor holds 25250ml so with the default calc that's 101 encumbrance. 250ml per encumbrance I think could be audited.
